test-data/unit/check-modules-case.test

```diff
@@ -0,0 +1,57 @@
+-- Type checker test cases dealing with modules and imports on case-insensitive filesystems.
+
+[case testCaseSensitivityDir]
+from a import B  # E: Module 'a' has no attribute 'B'
+
+[file a/__init__.py]
+[file a/b/__init__.py]
+
+[case testCaseInsensitivityDir]
+# flags: --config-file tmp/mypy.ini
+
+from a import B  # E: Module 'a' has no attribute 'B'
+from other import x
+reveal_type(x)  # E: Revealed type is 'builtins.int'
+
+[file a/__init__.py]
+[file a/b/__init__.py]
+[file FuNkY_CaSe/other.py]
+x = 1
+
+[file mypy.ini]
+[[mypy]
+mypy_path = tmp/funky_case
+
+[case testPreferPackageOverFileCase]
```

Comment:
Looks like the test name is backwards. But don't we also need a test for the reverse? (Where the package has the right case and so is preferred.)

Reply:
There is a similar test case `testPreferPackageOverFile` in `check-modules.test`. I will add however another one here with `mypy_path` just in case.